From 92101ea43dccf9afc9af34c02a30de381d3fe66a Mon Sep 17 00:00:00 2001
From: Gordon Ross <gwr@nexenta.com>
Date: Mon, 1 Jul 2019 16:00:34 -0400
Subject: [PATCH] 14097 time to disable SMB1 by default Reviewed by: Rick
 McNeal <rick.mcneal@nexenta.com> Reviewed by: Sanjay Nadkarni
 <sanjay.nadkarni@nexenta.com> Reviewed by: Matt Barden
 <matt.barden@nexenta.com> Reviewed by: Yuri Pankov <yuri.pankov@nexenta.com>
 Approved by: Dan McDonald <danmcd@mnx.io>

---
 usr/src/cmd/smbsrv/smbd/server.xml         |  2 +-
 usr/src/lib/smbsrv/libsmb/common/smb_cfg.c | 15 +++++++++------
 2 files changed, 10 insertions(+), 7 deletions(-)

diff --git a/usr/src/cmd/smbsrv/smbd/server.xml b/usr/src/cmd/smbsrv/smbd/server.xml
index 304887f1dd..cbf6f34d7c 100644
--- a/usr/src/cmd/smbsrv/smbd/server.xml
+++ b/usr/src/cmd/smbsrv/smbd/server.xml
@@ -234,7 +234,7 @@ file.
 		<propval name='max_protocol' type='astring'
 			value='' override='true'/>
 		<propval name='min_protocol' type='astring'
-			value='' override='true'/>
+			value='2.1' override='true'/>
 		<propval name='encrypt' type='astring'
 			value='disabled' override='true'/>
 		<propval name='encrypt_ciphers' type='astring'
diff --git a/usr/src/lib/smbsrv/libsmb/common/smb_cfg.c b/usr/src/lib/smbsrv/libsmb/common/smb_cfg.c
index 76441464ff..0d084fb556 100644
--- a/usr/src/lib/smbsrv/libsmb/common/smb_cfg.c
+++ b/usr/src/lib/smbsrv/libsmb/common/smb_cfg.c
@@ -1208,11 +1208,14 @@ smb_config_get_protocol(smb_cfg_id_t id, char *name, uint32_t default_val)
 }
 
 /*
- * The service manifest has empty values by default for min_protocol and
- * max_protocol. The expectation is that when those values are empty, we don't
- * constrain the range of supported protocol versions (and allow use of the
- * whole range that we implement). For that reason, this should usually be the
- * highest protocol version we implement.
+ * If the service properties min_protocol and max_protocol are empty values,
+ * the built-in defaults allow clients to use any supported protocol version.
+ * Policy choices (such as "let's disable SMB1") should be implemented by
+ * setting values in the SMB service, either via the service manifest (if a
+ * distribution policy) or via svccfg/svcprop (if a local policy).
+ *
+ * max_protocol_default should be the highest implemented protocol version.
+ * See also $UTS/common/fs/smbsrv/smb2_negotiate.c
  */
 uint32_t max_protocol_default = SMB_VERS_3_11;
 
@@ -1228,7 +1231,7 @@ smb_config_get_max_protocol(void)
 }
 
 /*
- * This should eventually be SMB_VERS_2_BASE
+ * See comment above max_protocol_default
  */
 uint32_t min_protocol_default = SMB_VERS_1;
 
-- 
2.46.0

